pkg/gitprovider/gitlab_test.go

```diff
@@ -43,6 +43,50 @@ func (g *GitLabGitProviderTestSuite) TestParseStaticGitContext_HTTP() {
 	require.Equal(httpContext, simpleContext)
 }
 
+func (g *GitLabGitProviderTestSuite) TestParseStaticGitContextSHA() {
```

Comment:
You'll notice that we have `TestParseStaticGitContext_Commits` below. Please just update that test with one more test case for the SHA scenario rather than introduce 2 new tests.